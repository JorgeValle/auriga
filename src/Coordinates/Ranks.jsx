import React from 'react';

// Components should represent one piece of data model
class Ranks extends React.Component {

  // It only has a render method, because this is a static version of app
  // To make UI interactive, we need to be able to trigger changes to underlying data model
  render() {
    return (
      <section className="ranks">
        <div className="rank">
          <span className="rank-letter">1</span>
        </div>
        <div className="rank">
          <span className="rank-letter">2</span>
        </div>
        <div className="rank">
          <span className="rank-letter">3</span>
        </div>
        <div className="rank">
          <span className="rank-letter">4</span>
        </div>
        <div className="rank">
          <span className="rank-letter">5</span>
        </div>
        <div className="rank">
          <span className="rank-letter">6</span>
        </div>
        <div className="rank">
          <span className="rank-letter">7</span>
        </div>
        <div className="rank">
          <span className="rank-letter">8</span>
        </div>
      </section>
    );
  }
}

export default Ranks;