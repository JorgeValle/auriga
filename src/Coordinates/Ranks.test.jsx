import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Ranks from './Ranks.jsx';

configure({
  adapter: new Adapter()
});

describe('<Ranks />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<Ranks />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<Ranks />);
    expect(wrapper).toMatchSnapshot();
  });

});

