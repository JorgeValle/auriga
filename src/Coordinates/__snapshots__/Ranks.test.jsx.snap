// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ranks /> matches snapshot 1`] = `ShallowWrapper {}`;
